kamon-core/src/main/scala/kamon/trace/Tracer.scala

```diff
@@ -158,6 +159,11 @@ object Tracer {
       this
     }
 
+    def withTraceID(identifier: IdentityProvider.Identifier): SpanBuilder = {
```

Comment:
what if also we add  withSpanID(...)?

Reply:
We could add this, but there is one little thing that we should take into account here: we have the "join with parent span id" thingy which is basically done only for compatibility with Zipkin.. Take a look at https://github.com/kamon-io/kamon-akka-remote/issues/15 for more context.

I'm going to do some tests with the very latest Zipkin, I have this vague idea that it would just work if client/server side had different span ids. Will also ask in their gitter.

Reply:
Thanks for the info, Adrian!

One thing that I was thinking to add was some sort of pre/post processors for Spans and maybe we can stay doing the right thing for Zipkin, but only on the spans with `span.kind=server` without it being a big exception in the main span builder path. 

The main motivation for the pre-processors was the ability to copy certain tags from the context into the Span... that is something that this PR is bringing but only for the HTTP server spans and if there is some sort of correlation ID or customer ID, that's most likely something you want to add to all spans, not just the server side. If we have a pre-processor for this, we can as well have a pre-processor for reusing the SpanID when "zipkin mode" is on. I'll play a bit with this idea and see how it goes.

On the post-processor side, I remember having a conversation with Bogdan from OC back in May where he suggested moving metrics collection to some sort of post-finish hook on the Spans which might make things simpler and easier to extend around Kamon's tracer. Although, I wouldn't like to introduce such changes until we gather more use cases for it.